catroid/src/androidTest/java/org/catrobat/catroid/test/io/asynctask/CopyProjectToExternalStorageAsyncTaskTest.java

```diff
@@ -0,0 +1,115 @@
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.io.asynctask;
+
+import android.Manifest;
+import android.app.NotificationManager;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.rule.GrantPermissionRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.util.Pair;
+import android.util.SparseArray;
+
+import org.catrobat.catroid.ProjectManager;
+import org.catrobat.catroid.content.Project;
+import org.catrobat.catroid.content.Scene;
+import org.catrobat.catroid.io.StorageOperations;
+import org.catrobat.catroid.io.asynctasks.CopyProjectToExternalStorageAsyncTask;
+import org.catrobat.catroid.test.utils.Reflection;
+import org.catrobat.catroid.utils.NotificationData;
+import org.catrobat.catroid.utils.StatusBarNotificationManager;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+
+import static org.catrobat.catroid.common.Constants.CATROBAT_EXTENSION;
+import static org.catrobat.catroid.common.FlavoredConstants.EXTERNAL_STORAGE_ROOT_EXPORT_DIRECTORY;
+@RunWith(AndroidJUnit4.class)
+public class CopyProjectToExternalStorageAsyncTaskTest {
+
+	private String projectName;
+	private File externalProjectZip;
+
+	@Rule
+	public GrantPermissionRule runtimePermissionRule = GrantPermissionRule.grant(Manifest.permission.WRITE_EXTERNAL_STORAGE, android.Manifest.permission.READ_EXTERNAL_STORAGE);
+
+	@Before
+	public void setUp() {
+		Project project = new Project(InstrumentationRegistry.getTargetContext(), "testProject");
+		project.addScene(new Scene());
+		projectName = project.getName();
+		ProjectManager.getInstance().setCurrentProject(project);
+		ProjectManager.getInstance().saveProject(InstrumentationRegistry.getTargetContext());
+	}
+
+	@Test
+	public void exportProjectTest() throws InterruptedException {
+		StatusBarNotificationManager notificationManager = StatusBarNotificationManager.getInstance();
+		int notificationID = notificationManager
+				.createSaveProjectToExternalMemoryNotification(InstrumentationRegistry.getTargetContext(), projectName);
+		CopyProjectToExternalStorageAsyncTask asyncTask = new CopyProjectToExternalStorageAsyncTask();
```

Comment:
no.
do not start async tasks in tests. (especially if you just have the test fail if it doesnt return quick enough without stopping the task...)

copyProjectToExternalStorage is annotated as @VisibleForTesting, you can just assert the very same thing in fewer lines when just directly calling copyProjectToExternalStorage and asserting that it behaves correctly without starting the task.

Reply:
👍